HHS/simpler-grants-gov
# [Issue #2695] opportunity and search toggle with feature flag env vars

## Summary
Fixes #2695

### Time to review: __30 mins__

## Changes proposed
The goal of the PR is the addition of two "kill switch" parameters that can be used to direct the search and opportunity pages, respectively, to a maintenance page. These parameters can be supplied either on the frontend via query params or feature flags admin panel, or via environment variables.

Setting flags via environment variables is a new functionality, so changes were made to the system to get this to work. Some deficiencies of the existing system were also patched up as part of this process.

## Context for reviewers
### Test Steps

1. `FEATURE_OPPORTUNITY_OFF=true FEATURE_SEARCH_OFF=true npm run dev` on this branch
2. visit http://localhost:3000/search
3. _VERIFY_: maintenance page appears
4. _VERIFY_: maintenance page matches designs and includes correct text from spec
2. visit http://localhost:3000/opportunity/1
3. _VERIFY_: maintenance page appears
4. _VERIFY_: maintenance page matches designs and includes correct text from spec
2. visit http://localhost:3000/search?_ff=searchOff%3Afalse
3. _VERIFY_: search page appears
4. 2. visit http://localhost:3000/opportunity/1?_ff=opportunityOff%3Afalse
3. _VERIFY_: opportunity page appears
8. Repeat 2 - x but in faux prod mode with `npm run build && FEATURE_OPPORTUNITY_OFF=true FEATURE_SEARCH_OFF=true npm start`

#### Dockerized
I would like to be able to test the changes to the Dockerfile locally, but can't without creating a docker compose that runs the production Next build and start rather than `next dev`. I feel like that's out of scope for this change, so we should just deploy out to DEV and test there.

## Additional information
> Screenshots, GIF demos, code examples or output to help show the changes working as expected.
